tests/unit/s2n_tls13_hybrid_shared_secret_test.c

```diff
@@ -374,9 +382,10 @@ static int set_up_conns(struct s2n_connection *client_conn, struct s2n_connectio
     client_conn->secure.server_kem_group_params.kem_params.kem = kem_group->kem;
     client_conn->secure.chosen_client_kem_group_params->kem_params.kem = kem_group->kem;
 
-    /* PQ shared secret is stored in the server_kem_group_params struct, regardless if connection is client or server */
+    /* During an actual handshake, server will generate the shared secret and store it in server_kem_group_params,
+     * client will decapsulate the ciphertext and store the shared secret in chosen_client_kem_group_params. */
     GUARD(s2n_dup(pq_shared_secret, &server_conn->secure.server_kem_group_params.kem_params.shared_secret));
```

Comment:
I noticed you changed line 388 from `.` to `->`, curious why the same doesn't hold true for the server side? 

Should this be `server_conn->secure.server_kem_group_params->kem_params.shared_secret`?

Reply:
Like classic ECDHE, the `kem_group_params` for both client and server are stored in the connection state; the client has an array of `kem_group_params` so that multiple key shares can be sent/received. `struct s2n_kem_group_params *chosen_client_kem_group_params` is a pointer that points to the `kem_group_params` element in the array that is chosen by the server for negotiation. (It's simply for convenience of access, so that we don't have to loop through the array and compare IANA values when we want to find the client key share for negotiation.) On the other hand, the server doesn't require an array/pointer.

https://github.com/awslabs/s2n/blob/4d667edcccf65a7f44dea8fdb1ec95b7db29b125/tls/s2n_crypto.h#L40-L42